thewalrus/tests/test_entanglement.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Entanglement tests"""
+# pylint: disable=no-self-use, assignment-from-no-return
+import pytest
+
+import numpy as np
+
+from thewalrus.quantum import entanglement_entropy, log_negativity
+from thewalrus.random import random_interferometer
+from thewalrus.symplectic import (
+    interferometer,
+    passive_transformation,
+    two_mode_squeezing,
+    xxpp_to_xpxp,
+)
+
+
+def random_cov(num_modes=200, pure=False, nonclassical=True):
+    r"""Creates a random covariance matrix for testing.
+
+    Args:
+        num_modes (int): number of modes
+        pure (bool): Gaussian state is pure
+        nonclassical (bool): Gaussian state is nonclassical
+
+    Returns:
+        (array): a covariance matrix
+    """
+    M = num_modes
+    O = interferometer(random_interferometer(M))
+
+    if pure:
+        eta = 1
+    elif not pure:
+        eta = 0.123
+
+    r = 1.234
+    if nonclassical:
+        # squeezed inputs
+        cov_in = np.diag(np.concatenate([np.exp(2 * r) * np.ones(M), np.exp(-2 * r) * np.ones(M)]))
+    elif not nonclassical and pure:
+        # vacuum inputs
+        cov_in = np.eye(2 * M)
+    elif not nonclassical and not pure:
+        # squashed inputs
```

Comment:
Why remove the explicit `elif` statement here? I guess it's a convention thing? Doesn't it help a reader to distinguish the cases better?

Reply:
That's true. This is more of a Python convention thing, since the final check is unnecessary. There might be slight differences such as quicker evaluations and more precise type hint checks (it's potentially easier for a language server to check if at least one case is run if there's an `else`), but probably nothing major. If you think it's clearer to have the final `elif` then I think it's fine to keep it.

Reply:
If you really don't mind I'd slightly prefer to stick to the original version here, simply because it makes the code easier to understand -- even for myself.

That being said, this function might have a limited life time anyway as @nquesada put forward the idea of upgrading the existing random module so it accepts additional inputs like `pure`, `classical`, etc. In this case, this function here will become redundant.

Reply:
Sounds good! I'm alright with that! 👍